File: python/char_count.py
def char_count(str):
  str = str.replace(' ', '')
  result = []
  for char in list(str):
    if not any(char in elem for elem in result):
      result.append([char, 1])
    else:
      for elem in range(0,len(result)):
        if result[elem][0] == char:
          result[elem][1] += 1

  sorted_result = sorted(result, key=lambda x: (-x[1], x[0]))

  #sorted_result = sorted(result, key=sort_result)
  # need to sort nested array by second index
    
  return sorted_result
  

# print(char_count("aaa  bbc"))

# print(char_count("an apple a day will keep the doctor away"))

File: js/charCount.js
exports.charCount = function(str) {
    str = str.replaceAll(' ', '');
    let countObj = {};
    let result = [];

    for (let i = 0; i < str.length; i++) {
        if (str[i] in countObj === false) {
            countObj[str[i]] = 1;
        } else {
            countObj[str[i]] += 1;
        }
    }
    
    objLength = Object.entries(countObj).length;
    for (let i = 0; i < objLength; i++) {
        result[i] = Object.entries(countObj)[i];
    }

    let sortedResult = result.sort((a,b) => {
        return a[1] === b[1] ? a[0].charCodeAt(0) - b[0].charCodeAt(0) : b[1] - a[1]
    })

    return sortedResult
};